### PR TITLE
CEDS-944 CEDS-945 Item Type multiplicity

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

**Comment by PauloPortugal** [comment]
I really like your commit hygiene :)

File: app/controllers/supplementary/ItemTypePageController.scala [hunk 1]
```diff
@@ -42,21 +46,139 @@ class ItemTypePageController @Inject()(
...
+    ItemType(
+      combinedNomenclatureCode = itemTypeInput.combinedNomenclatureCode,
+      taricAdditionalCodes = itemTypeCache.taricAdditionalCodes ++ itemTypeInput.taricAdditionalCodes,
+      nationalAdditionalCodes = itemTypeCache.nationalAdditionalCodes ++ itemTypeInput.nationalAdditionalCodes,
+      descriptionOfGoods = itemTypeInput.descriptionOfGoods,
+      cusCode = itemTypeInput.cusCode,
+      statisticalValue = itemTypeInput.statisticalValue
+    )
+
+  private def validateItemType(itemType: ItemType)(implicit request: Request[AnyContent]): Future[ValidationResult] =
```
Comment (PatrykRudnicki) [inline app/controllers/supplementary/ItemTypePageController.scala]: You can get rid of this method or pass action type. You use this method inside `handleAddition` and `handleSaveAndContinue` so you know what type you have. Based on it you don't need to use `getOrElse`

File: app/forms/supplementary/ItemType.scala [hunk 2]
```diff
@@ -53,74 +63,58 @@ case class ItemType(
...
-        "supplementary.itemType.description.error.length",
-        isEmpty or noLongerThan(descriptionOfGoodsMaxLength)
-      ),
-    "cusCode" -> optional(
-      text()
-        .verifying("supplementary.itemType.cusCode.error.length", hasSpecificLength(cusCodeLength))
-        .verifying("supplementary.itemType.cusCode.error.specialCharacters", isAlphanumeric)
-    ),
+    "combinedNomenclatureCode" -> text(),
+    taricAdditionalCodesKey -> default(seq(text()), Seq.empty),
```
Comment (PatrykRudnicki) [inline app/forms/supplementary/ItemType.scala]: Why this? `default(seq(text()), Seq.empty)`

File: app/controllers/supplementary/ItemTypePageController.scala (hunk 1 above)
Comment (MaciejRewera) [reply app/controllers/supplementary/ItemTypePageController.scala]: Removed

File: app/forms/supplementary/ItemType.scala (hunk 2 above)
Comment (MaciejRewera) [reply app/forms/supplementary/ItemType.scala]: Because without default() we would bind empty string into the sequence.

**Pull request merged**